### PR TITLE
vai pagination tests - CustomResourceDefinitions

### PR DESCRIPTION
**Pull request by yonasberhe23**
### Summary
Fixes #https://github.com/rancher/qa-tasks/issues/1380
Covers pagination functionality for resource list under CustomResourceDefinitions.

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [ ] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

**Comment by richard-cox** [comment]
Pausing on review until Tokens PR has merged

**Comment by yonasberhe23** [comment]
> Pausing on review until Tokens PR has merged

@richard-cox this is ready for ready for review

File: cypress/e2e/po/pages/explorer/custom-resource-definitions.po.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import PagePo from '@/cypress/e2e/po/pages/page.po';
+import BaseResourceList from '@/cypress/e2e/po/lists/base-resource-list.po';
+import BurgerMenuPo from '@/cypress/e2e/po/side-bars/burger-side-menu.po';
+import ProductNavPo from '@/cypress/e2e/po/side-bars/product-side-nav.po';
+import CodeMirrorPo from '@/cypress/e2e/po/components/code-mirror.po';
+import CustomResourceDefinitionsCreateEditPo from '@/cypress/e2e/po/edit/custom-resource-definitions.po';
+
+export class CustomResourceDefinitionsPagePo extends PagePo {
```
Comment (richard-cox) [inline cypress/e2e/po/pages/explorer/custom-resource-definitions.po.ts:8]: Comment only. There's a lot of boilerplate / c&p now for resource list pages.

At some point in the future we should create an abstract class ResourceList that these type of pages can extend, then override specific things like createPath, navTo, etc

File: cypress/e2e/tests/pages/explorer/more-resources/api/custom-resource-definitions.spec.ts [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        const json: any = jsyaml.load(crdYaml);
+
+        json.metadata.name = crdName;
+        json.spec.group = crdGroup;
+
+        crdsPage.yamlEditor().set(jsyaml.dump(json));
+      });
+
+      cy.intercept('POST', 'v1/apiextensions.k8s.io.customresourcedefinitions').as('createCRD');
+      crdsPage.crdCreateEditPo().saveCreateForm().resourceYaml().saveOrCreate()
```
Comment (richard-cox) [inline cypress/e2e/tests/pages/explorer/more-resources/api/custom-resource-definitions.spec.ts:32]: Comment only. This is nice as it tests the create from yaml process, however it's probably best placed in a specific test of its own and instead we create the resource via API request (there's an endpoint that can handle yaml used by the `Import YAML` header button/modal)

**Review by richard-cox — APPROVED** [review]
There's a couple of general things we can address in the future. In the interest of time (situation caused by the reviewer who took a horrible amount of time to review) lets get this in

File: cypress/e2e/tests/pages/explorer/more-resources/api/custom-resource-definitions.spec.ts (hunk 2 above)
Comment (yonasberhe23) [reply cypress/e2e/tests/pages/explorer/more-resources/api/custom-resource-definitions.spec.ts:32]: ah i was wondering this. thanks for calling it out. I'll dm you if i cant get it working

**Pull request merged**